### PR TITLE
fix(theme): --vp-nav-height should be 0 when navbar is hidden

### PR DESCRIPTION
**Pull request by buqiyuan**
before：
image

after：

**If this PR is merged, I will add document for it later.**

### CODE REVIEW

File: src/client/theme-default/components/VPNav.vue [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
 
 provide('close-screen', closeScreen)
+
+watchEffect(() => {
+  if (hasNavbar.value) {
+    document.documentElement.style.removeProperty('--vp-nav-height')
+  } else {
+    document.documentElement.style.setProperty('--vp-nav-height', '0')
+  }
+})
```
Comment (brc-dd) [inline src/client/theme-default/components/VPNav.vue:23]: this will throw errors during build

Comment (buqiyuan) [reply src/client/theme-default/components/VPNav.vue]: Yeah, you are right

**Comment by itsWindows11** [comment]
In terms of design I'm not particularly a fan of this change, could add some spacing at the top (like 8 or 12px margin as nav height).

**Comment by kiaking** [comment]
yeah it should have some spacing. Can this be fixed if we use `position: sticky` for navbar instead of `position: fixed`? 🤔

**Pull request merged**